Documentation/Usage.md

```diff
@@ -397,3 +398,34 @@ manifest API).
 It is *not* expected the packages would ever use this feature unless absolutely
 necessary to support existing clients. In particular, packages *should not*
 adopt this syntax for tagging versions supporting the _latest GM_ Swift version.
+
+### Editable Packages
+
+SwiftPM supports editing dependencies, which means moving it into a location under `Packages/` directory. 
+
+If a such an editable package is present in Packages, then swift build will always use the exact sources in this directory to build, regardless of it's state, git repository status, tags, or the tag desired by dependency resolution. In other words, this will "just build" against the sources that are present.
+
+When an editable package is present, it will be used to satisfy all instances of that Package in the depencency graph. It is possible to edit all, some, or none of the packages in a dependency graph, without restriction.
+
+Editable package are best used to do experimentation with dependency code or create and submit a patch in the dependency owner's repository (upstream).
+
+There are two ways to put a package in editable state:
+
+    $ swift package edit Foo --branch bugFix
+
+This will create a branch called `bugFix` from currently resolved version and put the dependency Foo in Packages/ directory. 
+
+    $ swift package edit Foo --revision 969c6a9
+
+This is similar to previous version except that SwiftPM will leave the dependency at a detched HEAD on the specified revision.
+
+Once a package is in editable state, you can navigate to the directory `Packages/Foo` make changes, build and then push the changes to the upstream repository.
+
+You can end editing a package using `unedit` command:
+
+    $ swift package unedit Foo # --force
+
+This will remove the edited dependency from Packages/ and put the originally resolved version back. This command will fail if there are uncommited changes or changes which are not
+pushed to the remote repository. The `--force` option can be used to forcefully unedit a dependency.
```

Comment:
Should probably clarify that this will discard the edits (assuming that's what it does).